server.py

```diff
@@ -0,0 +1,85 @@
+from __future__ import unicode_literals
+import socket
+
+
+_CRLF = b'\r\n'
+_GET = b'GET'
+_PROTOCOL = b'HTTP/1.1'
+_HOST_PREFIX = b'Host:'
+_WS = b' '
+
+_RESPONSE_TEMPLATE = _CRLF.join([
+    b'HTTP/1.1 {response_code} {response_reason}',
+    b'Content-Type: text/html; charset=UTF-8',
+    b''])
+
+
+def response_ok():
+    return _RESPONSE_TEMPLATE.format(response_code=b'200',
+                                     response_reason=b'OK')
+
+
+def response_error(response_code, response_reason):
+    """ Return 500 Internal Server Error"""
+    return _RESPONSE_TEMPLATE.format(response_code=response_code,
+                                     response_reason=response_reason)
+
+
+def parse_request(request):
+    lines = request.split(_CRLF)
+
+    # Validate the header line
+    header = lines[0]
+    header_pieces = header.split(_WS)
```

Comment:
as discussed in class, don't split on a single space character when what you mean is to split on "whitespace".  Just leave the argument out of split and it will treat any number of spaces, tabs, etc as "whitespace"